cause grunt to fail on selector count about 4095

fluxusfrequency opened the pull request:
When doing a build, it is sometimes helpful to be able to fail out (e.g. for CI) if a CSS file is over the limit.

The auto-split behavior is nice, but not always appropriate. Since you sometimes need to manually update your templates or html, some people might prefer a stop-the-build mode of operation.

This PR adds an option called failOnError that will cause grunt to fail out when the selector count is over the limit.

patrickkettner commented:
@AaronAsAChimp any chance this can be added? Would be super useful

AaronAsAChimp commented:
This PR needs some work before it can be merged.

- Indentation must be 1 tab per indent
- New features must be documented in the README
- The error message should not be repeated
- The name of the option "failOnError" does not describe what it does. Something like "failOnLimit" would be better.

fluxusfrequency commented:
Thanks for taking the time to look this over, @AaronAsAChimp. I pushed another commit that addresses your concerns. Please let me know if there's anything else you'd like to see.

AaronAsAChimp commented:
Just one last thing, the lines you added in this last commit need to be tabs, thats it! I'll merge this when you're done.

patrickkettner commented:
if I could make one suggestion - it would be great to not go `fatal` on first error, in case you are parsing multiple files that have the issue.

AaronAsAChimp commented:
@patrickkettner I would say thats a nice-to-have, but shouldn't hold up this PR.
